includes/class-wc-gateway-stripe.php

```diff
@@ -1015,7 +1034,44 @@ public function failed_sca_auth( $order, $intent ) {
 			? sprintf( __( 'Stripe SCA authentication failed. Reason: %s', 'woocommerce-gateway-stripe' ), $intent->last_payment_error->message )
 			: __( 'Stripe SCA authentication failed.', 'woocommerce-gateway-stripe' );
 		$order->update_status( 'failed', $status_message );
+	}
+
+	/**
+	 * Given a response from Stripe, check if it's a card error where authentication is required
+	 * to complete the payment.
+	 *
+	 * @param object The response from Stripe.
+	 * @return boolean Whether or not it's a 'needs_authentication' error
+	 */
+	public function is_authentication_required_for_payment( $intent ) {
+		return ( ! empty( $intent->error ) && 'authentication_required' === $intent->error->code )
+			|| ( ! empty( $intent->last_payment_error ) && 'authentication_required' === $intent->last_payment_error->code );
+	}
 
-		$this->send_failed_order_email( $order->get_id() );
+	public function process_authentication_required_response( $order ) {
+		$auth_url = add_query_arg( 'wc-stripe-confirmation', 1, $order->get_checkout_payment_url( false ) );
+		$order->add_order_note( '[TODO: send a real e-mail here] Go here to authenticate: ' . $auth_url, true );
```

Comment:
@RadoslavGeorgiev If we keep these changes, you'll need to replace this `add_order_note` with sending a real e-mail with the `$auth_url` link.